test/mlir/krnl/krnl_getref_lowering.mlir

```diff
@@ -10,21 +10,19 @@ func @test_getref_lowering(%arg0: memref<2x2xf32>) -> memref<2x2xf32> {
   // CHECK: %[[OFFSET:.+]] = llvm.mlir.constant(13 : i64) : !llvm.i64
   // CHECK: [[CONST_10_0:%.+]] = llvm.mlir.constant(10 : index) : !llvm.i64
   // CHECK: [[CONST_10_1:%.+]] = llvm.mlir.constant(10 : index) : !llvm.i64
-  // CHECK: [[MUL1:%.+]] = llvm.mul [[CONST_10_0]], [[CONST_10_1]] : !llvm.i64
+  // CHECK: [[CONST_1:%.+]] = llvm.mlir.constant(1 : index) : !llvm.i64
+  // CHECK: %[[CONST_100:.+]] = llvm.mlir.constant(100 : index) : !llvm.i64
   // CHECK: [[FLOAT_STAR:%.+]] = llvm.mlir.null : !llvm.ptr<float>
-  // CHECK: %[[CONST_1:.+]] = llvm.mlir.constant(1 : index) : !llvm.i64
-  // CHECK: [[ELEM1:%.+]] = llvm.getelementptr [[FLOAT_STAR]][%[[CONST_1]]] : (!llvm.ptr<float>, !llvm.i64) -> !llvm.ptr<float>
+  // CHECK: [[ELEM1:%.+]] = llvm.getelementptr [[FLOAT_STAR]][%[[CONST_100]]] : (!llvm.ptr<float>, !llvm.i64) -> !llvm.ptr<float>
   // CHECK: [[ELEM_SIZE:%.+]] = llvm.ptrtoint [[ELEM1]] : !llvm.ptr<float> to !llvm.i64
-  // CHECK: [[MUL2:%.+]] = llvm.mul [[MUL1]], [[ELEM_SIZE]] : !llvm.i64
-  // CHECK: [[MEMPOOL:%.+]] = llvm.call @malloc([[MUL2]]) : (!llvm.i64) -> !llvm.ptr<i8>
+  // CHECK: [[MEMPOOL:%.+]] = llvm.call @malloc([[ELEM_SIZE]]) : (!llvm.i64) -> !llvm.ptr<i8>
```

Comment:
Looks like we are missing a multiplier here. ELEM_SIZE needs to be multiplied by the number of elements. How did this change come about?

Reply:
This and the other malloc do look wrong. Let me try to figure out what's going on here. Thanks!

Reply:
This appears to be correct now, after https://reviews.llvm.org/D91549 (new behvaior for getMemRefDescriptorSizes, now returning sizeBytes directly and using constants (see CONST_100 in llvm.getelemptr call above)). I'll add it to the list of LLVM changes related to this patch. You can see the change to the identical pattern in line 175 of convert-stic-memref-ops.mlir, here https://github.com/llvm/llvm-project/commit/bedaad44953a99138bda0ee08ceeb5dfeec3c35c#diff-6bc2cec081b14c8e9e74b632e3f4d92ba4055334dde7a0fa7b6c43ea1b42d520). I believe this should allow you to resolve this and the other related reviews, but please let me know if anything is outstanding.

Reply:
I see what's happening now. This makes sense.

Reply:
Thanks a lot for looking into this.